From: Johan Hedberg <johan.hedberg@nokia.com>
Date: Mon, 8 Jun 2009 04:06:33 +0000 (+0700)
Subject: Fix bt_cancel_discovery calls in case of SDP failures
X-Git-Url: http://git.kernel.org/?p=bluetooth%2Fbluez.git;a=commitdiff_plain;h=9d7f87e0e8693b3645da5d66dec736d1ee3e9359

Fix bt_cancel_discovery calls in case of SDP failures

We shouldn't call bt_cancel_discovery if we're withing a SDP callback so
always make sure that p->svclass is 0 before calling
pending_connect_finalize (which uses this value to determine whether
cancelation is needed). The glib-helper.c code should still safeguard
against this too and that's what the previous commit fixes.
---

diff --git a/audio/headset.c b/audio/headset.c
index 7e689ff..86cce52 100644
--- a/audio/headset.c
+++ b/audio/headset.c
@@ -1456,6 +1456,7 @@ failed_not_supported:
 	if (p->msg)
 		error_not_supported(dev->conn, p->msg);
 failed:
+	p->svclass = 0;
 	pending_connect_finalize(dev);
 	headset_set_state(dev, HEADSET_STATE_DISCONNECTED);
 }
